presto-main/src/main/java/com/facebook/presto/sql/gen/ExpressionCompiler.java

```diff
@@ -210,4 +256,55 @@ public String toString()
                     .toString();
         }
     }
+
+    private static final class JoinFilterCacheKey
+    {
+        private final RowExpression filter;
+        private final int leftBlocksSize;
+
+        public JoinFilterCacheKey(RowExpression filter, int leftBlocksSize)
+        {
+            this.filter = filter;
+            this.leftBlocksSize = leftBlocksSize;
```

Comment:
This should be true now as we do not use compiled join-filter functions this for CROSS JOIN with condition. But it is not strong requirement for this class so I do not feel that we want to enforce it here.